lib/srv/forward/sshserver.go

```diff
@@ -1153,10 +1153,15 @@ func (s *Server) handleEnv(ch ssh.Channel, req *ssh.Request, ctx *srv.ServerCont
 
 func (s *Server) replyError(ch ssh.Channel, req *ssh.Request, err error) {
 	s.log.Error(err)
-	message := utils.UserMessageFromError(err)
-	s.stderrWrite(ch, message)
+	message := utils.FormatError(err)
+	// Terminate the error with a newline when writing to remote channel's
+	// stderr so the output does not mix with the rest of the output if the remote
+	// side is not doing additional formatting for extended data.
+	// See github.com/gravitational/teleport/issues/4542
+	payload := message + "\n"
```

Comment:
Would it make sense to append `\n` only if `message` doesn't already end with it? Or it never ends with a newline?

Reply:
I don't think this is customary to have error messages ending in a new line in teleport code base itself. This is meant for external SSH clients (actually a single client that has this behavior - OpenSSH).

Reply:
IMO we should still add the newline conditionally, to avoid any assumptions about out error messages and blank lines on openssh side